### PR TITLE
Allow docker hostconfig in configuration

### PR DESCRIPTION
**Pull request by fsaintjacques**
This gives control on setting cpuset with `cpuset_cpus` options, and
eventually maximum memory.

### CODE REVIEW

**Comment by fsaintjacques** [comment]
There's going to be a followup PR to limit dgx2 to benchmarks only.

File: master.cfg [hunk 1]
```diff
@@ -186,7 +186,8 @@ for worker in conf.workers:
             image=util.Property('docker_image'),
             autopull=True,
             alwaysPull=True,
-            hostconfig=dict(network_mode='host')
+            hostconfig=worker.hostconfig,
+            max_builds=worker.get('max_builds', 1)
```
Comment (kszucs) [inline master.cfg]: We should not limit `max_builds` by default.

Comment (fsaintjacques) [reply master.cfg]: I intend to refactor workers in prod.toml to have 1 worker per each set of 4-5 cpus.

File: test.toml [hunk 2]
```diff
@@ -16,8 +16,10 @@ database_url = 'sqlite:///ursabot.sqlite'
 name = 'local1'
 arch = 'amd64'
 docker = { host = 'unix://var/run/docker.sock' }
+hostconfig = { network_mode = 'host', cpuset_cpus = '0,1' }
```
Comment (kszucs) [inline test.toml]: Move these config keys under docker, because We'll have non-docker workers as well.

Comment (fsaintjacques) [reply test.toml]: Done

**Pull request merged**